pkg/kotsadmsnapshot/types/types_test.go

```diff
@@ -0,0 +1,144 @@
+package types
+
+import (
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+)
+
+func TestRollupStatus(t *testing.T) {
+	tests := []struct {
+		backupStatuses []BackupStatus
+		expected       BackupStatus
+	}{
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusInProgress,
+				BackupStatusInProgress,
+			},
+			expected: BackupStatusInProgress,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusInProgress,
+				BackupStatusFailed,
+			},
+			expected: BackupStatusInProgress,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusInProgress,
+				BackupStatusCompleted,
+			},
+			expected: BackupStatusInProgress,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusInProgress,
+				BackupStatusDeleting,
+			},
+			expected: BackupStatusInProgress,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusDeleting,
+				BackupStatusDeleting,
+			},
+			expected: BackupStatusDeleting,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusDeleting,
+				BackupStatusFailed,
+			},
+			expected: BackupStatusDeleting,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusDeleting,
+				BackupStatusCompleted,
+			},
+			expected: BackupStatusDeleting,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusFailed,
+				BackupStatusFailed,
+			},
+			expected: BackupStatusFailed,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusFailed,
+				BackupStatusCompleted,
+			},
+			expected: BackupStatusFailed,
+		},
+		{
+			backupStatuses: []BackupStatus{
+				BackupStatusCompleted,
+				BackupStatusCompleted,
+			},
+			expected: BackupStatusCompleted,
+		},
+	}
```

Comment:
This encodes the matrix states we talked about in the design doc here:
- https://docs.google.com/document/d/1JVgVSnrPWeLhS3qm3CC_MdDGDvFpM5TSABkySbHum-w/edit?tab=t.0#heading=h.qb8vv9xg5rhi